src/dyad/core/dyad_core.c

```diff
@@ -1030,41 +1029,45 @@ dyad_rc_t dyad_consume (dyad_ctx_t* ctx, const char* fname)
             rc = dyad_get_data (ctx, mdata, &file_data, &data_len);
             if (DYAD_IS_ERROR (rc)) {
                 DYAD_LOG_ERROR (ctx, "dyad_get_data failed!\n");
-                dyad_release_flock (ctx, fd, &exclusive_lock);
+                dyad_release_flock (ctx, lock_fd, &exclusive_lock);
                 goto consume_done;
             }
             DYAD_C_FUNCTION_UPDATE_INT ("data_len", data_len);
-
+            io_fd = open (fname, O_WRONLY);
```

Comment:
I am not sure about this. The lock is released before the file is written. So, there is a chance that another process initiates the metadata full and data full before this file is written.